controllers/menu_controller.rb

```diff
@@ -0,0 +1,128 @@
+require_relative "../models/address_book"
+
+class MenuController
+  attr_accessor :address_book
+
+  def initialize
+    @address_book = AddressBook.new
+  end
+
+  def main_menu
+
+    puts "Main Menu - #{@address_book.entries.count} entries"
+    puts "1 - View all entries"
+    puts "2 - Create an entry"
+    puts "3 - Search for an entry"
+    puts "4 - Import entries from a CSV"
+    puts "5 - View Entry \"n\""
+    puts "6 - Exit"
+    print "Enter your selection: "
+
+    selection = gets.to_i
+        case selection
+    when 1
+      system "clear"
+      view_all_entries
+      main_menu
+    when 2
+      system "clear"
+      create_entry
+      main_menu
+    when 3
+      system "clear"
+      search_entries
+      main_menu
+    when 4
+      system "clear"
+      read_csv
+      main_menu
+    when 5
+      system "clear"
+      view_entry
+      main_menu
+    when 6
+      puts "Good-bye!"
+      exit(0)
+      system "clear"
+      puts "Sorry, that is not a valid input"
+      main_menu
+    end
+  end
+
+  def view_all_entries
+    @address_book.entries.each do |entry|
+      system "clear"
+      puts entry.to_s
+      entry_submenu(entry)
+    end
+    system "clear"
+    puts "End of entries"
+  end
+
+  def create_entry
+    system "clear"
+    puts "New AddressBloc Entry"
+    print "Name: "
+    name = gets.chomp
+    print "Phone number: "
+    phone = gets.chomp
+    print "Email: "
+    email = gets.chomp
+    @address_book.add_entry(name, phone, email)
+    system "clear"
+    puts "New entry created"
+  end
+
+  def entry_submenu(entry)
+    puts "n - next entry"
+    puts "d - delete entry"
+    puts "e - edit this entry"
+    puts "m - return to main menu"
+    selection = gets.chomp
+    case selection
+    when "n"
+    when "d"
+    when "e"
+    when "m"
+      system "clear"
+      main_menu
+    else
+      system "clear"
+      puts "#{selection} is not a valid input"
+      entries_submenu(entry)
+    end
+  end
+
+  def view_entry
+    puts "Enter an index number"
+    print "Search: "
+    search = gets.chomp
+    @entries = @address_book.entries
```

Comment:
`@entries` is an instance variable which exposes it to the other methods in this instance. 

Try to minimize scope of variables and so minimize their exposure to methods that don't need to know about them. Over-exposed variables are more likely to get accidentally modified in other methods.

Consider changing `@entries` from an instance variable to a local variable. Local variables are only exposed in the method they're declared.